packages/core/src/client.ts

```diff
@@ -69,6 +68,41 @@ import { _getSpanForScope } from './utils/spanOnScope';
 const ALREADY_SEEN_ERROR = "Not capturing exception because it's already been captured.";
 const MISSING_RELEASE_FOR_SESSION_ERROR = 'Discarded session because of missing or non-string release';
 
+const INTERNAL_ERROR_SYMBOL = Symbol.for('SentryInternalError');
+const DO_NOT_SEND_EVENT_SYMBOL = Symbol.for('SentryDoNotSendEventError');
+
+interface InternalError {
+  message: string;
+  [INTERNAL_ERROR_SYMBOL]: true;
+}
+
+interface DoNotSendEventError {
+  message: string;
+  [DO_NOT_SEND_EVENT_SYMBOL]: true;
+}
+
+function makeInternalError(message: string): InternalError {
+  return {
+    message,
+    [INTERNAL_ERROR_SYMBOL]: true,
+  };
+}
+
+function makeDoNotSendEventError(message: string): DoNotSendEventError {
+  return {
+    message,
+    [DO_NOT_SEND_EVENT_SYMBOL]: true,
+  };
+}
+
+function isInternalError(error: unknown): error is InternalError {
+  return !!error && typeof error === 'object' && INTERNAL_ERROR_SYMBOL in error;
+}
+
+function isDoNotSendEventError(error: unknown): error is DoNotSendEventError {
+  return !!error && typeof error === 'object' && DO_NOT_SEND_EVENT_SYMBOL in error;
+}
```

Comment:
Although we want to differentiate them, we can use a single `isX` method for now given everything checks for both:

```ts
function isClientProcessingError(error: unknown): error is InternalError | DoNotSendEventError {
  return !!error && typeof error === 'object' && (INTERNAL_ERROR_SYMBOL in error || DO_NOT_SEND_EVENT_SYMBOL in error);
}
```

Reply:
actually it is not _entirely_ the same, today we differentiate based on this if we do `logger.log` or `logger.warn`. Not sure how important that is to us though 🤔

Reply:
I guess we can keep them separate for now then.